common/web/lm-worker/src/test/mocha/cases/suggestion-finalization.js

```diff
@@ -0,0 +1,418 @@
+import * as wordBreakers from '@keymanapp/models-wordbreakers';
+import { deepCopy } from '@keymanapp/web-utils';
+import { assert } from 'chai';
+
+import { finalizeSuggestions } from "#./predict-helpers.js";
+import { DummyModel } from "#./models/dummy-model.js";
+
+/*
+ * This file's tests use these parts of a lexical model:
+ * - model.wordbreaker
+ * - model.punctuation
+ */
+
+const testModelWithSpacing = new DummyModel({
+  wordbreaker: wordBreakers.default,
+  punctuation: {
+    quotesForKeepSuggestion: {
+      open: '<',
+      close: '>'
+    },
+    insertAfterWord: '\u00a0' // non-breaking space
+  }
+});
+
+const testModelWithoutSpacing = new DummyModel({
+  wordbreaker: wordBreakers.default,
+  punctuation: {
+    quotesForKeepSuggestion: {
+      open: '<',
+      close: '>'
+    },
+    insertAfterWord: ''
+  }
+});
+
+/**
+ * Builds a fresh copy of test values useful for suggestion-similarity
+ * testing.
+ * @returns
+ */
+const build_its_is_set = () => {
+  /** @type {import('#./predict-helpers.js').CorrectionPredictionTuple} */
+  const its = {
+    correction: {
+      sample: 'its',
+      p: 0.8
+    },
+    prediction: {
+      sample: {
+        transform: {
+          insert: 's',
+          deleteLeft: 0
+        },
+        displayAs: 'its'
+      },
+      p: 0.2
+    },
+    totalProb: 0.16
+    // matchLevel does not yet exist.
+  };
+
+  /** @type {import('#./predict-helpers.js').CorrectionPredictionTuple} */
+  const it_is = {
+    correction: {
+      sample: 'its',
+      p: 0.8
+    },
+    prediction: {
+      sample: {
+        transform: {
+          insert: '\'s',
+          deleteLeft: 0
+        },
+        displayAs: 'it\'s'
+      },
+      p: 0.8
+    },
+    totalProb: 0.64
+  };
+
+  /** @type {import('#./predict-helpers.js').CorrectionPredictionTuple} */
+  const is = {
+    correction: {
+      sample: 'is',
+      p: 0.2
+    },
+    prediction: {
+      sample: {
+        transform: {
+          insert: 's',
+          deleteLeft: 1
+        },
+        displayAs: 'is'
+      },
+      p: 0.5
+    },
+    totalProb: 0.1
+  };
+
+  /** @type {import('#./predict-helpers.js').CorrectionPredictionTuple} */
+  const is_not = {
+    correction: {
+      sample: 'is',
+      p: 0.2
+    },
+    prediction: {
+      sample: {
+        transform: {
+          insert: 'sn\'t',
+          deleteLeft: 1
+        },
+        displayAs: 'isn\'t'
+      },
+      p: 0.5
+    },
+    totalProb: 0.1
+  };
+
+  return {
+    its,
+    it_is,
+    is,
+    is_not
+  }
+};
+
+describe('finalizeSuggestions', () => {
+  describe('with custom post-token insert', () => {
+    it('with caret at end of current token and context', () => {
```

Comment:
the phrase within `it()` should describe the expected outcome, and make sense if "it" is prepended, something like this:

```suggestion
    it('should provide appropriate suggestions with caret at end of current token and context', () => {
```

For this and all `it()` calls in this module.